### PR TITLE
Add [FlakeHub] service

### LINKED ISSUE #9551: flakehub.com
### 📋 Description

There's a new service https://flakehub.com/ to add SemVer support for Nix flakes.

It would be great to have a badge that displayed the latest version published in that site.

### 🔗 Data

There seems to be an API, because the public site is actually using it to display its content. Examples:

- https://api.flakehub.com/flakes
- https://api.flakehub.com/f/copier-org/copier/releases

I can't find any docs. I opened https://github.com/DeterminateSystems/fh/issues/24 about this.

### 🎤 Motivation

Encouraging installing packages through stable releases in nix flake format.

### PR DESCRIPTION
**Pull request by lucperkins**
Fixes #9551. This is marked as a draft because the endpoint doesn't yet exist on FlakeHub but we're working on it 😄

### CODE REVIEW

**Comment by lucperkins** [comment]
We recently added support for badges on our side and would love to provide our users with official support 😄 https://api.flakehub.com/f/DeterminateSystems/fh/badge

File: services/flakehub/flakehub.tester.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    nock('https://api.flakehub.com')
+      .get('/f/DeterminateSystems/nuenv/badge')
+      .reply(200, {
+        latest: '0.1.160',
+      }),
+  )
+  .expectBadge({ label: 'FlakeHub', message: '0.1.160' })
+
+t.create('FlakeHub (not found)')
+  .get('/f/foo/bar')
```
Comment (cole-h) [inline services/flakehub/flakehub.tester.js:24]:
```suggestion
  .get('/f/DeterminateSystems/flakehub-this-will-never-exist-we-promise')
```

**Comment by chris48s** [comment]
Before reviewing the code in this PR, any chance of a follow up on https://github.com/badges/shields/issues/9551#issuecomment-1713814549 ?

**Comment by lucperkins** [comment]
@chris48s Ah, I completely forgot about that. I'll take a look!

**Comment by lucperkins** [comment]
We've managed to get this working without needing a direct integration here, so I'm going to close this. Such a cool services 😄

**Pull request closed without merging**